File: backend/app/schemas/customer.py
"""客户相关 Schema。"""
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class CustomerCreate(BaseModel):
    name: str
    industry: str | None = None
    phone: str | None = None
    email: str | None = None
    status: str = "private"
    remark: str | None = None


class CustomerOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    industry: str | None = None
    phone: str | None = None
    email: str | None = None
    status: str
    owner_id: int | None = None
    last_follow_up_at: datetime | None = None
    created_at: datetime
    updated_at: datetime

File: backend/src/main/java/com/crm/entity/Customer.java
package com.crm.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 客户。
 */
@Data
@TableName("customers")
public class Customer {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String name;
    private String industry;
    private String phone;
    private String email;
    private String status;
    private Long ownerId;
    private Long leadId;
    private LocalDateTime lastFollowUpAt;
    private String remark;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
